yarn-project/archiver/src/archiver/kv_archiver_store/message_store.ts

```diff
@@ -1,220 +1,91 @@
       if (lastL1BlockNumber >= lastMessageL1BlockNumber) {
         return false;
       }
 
-      void this.#lastL1BlockNewMessages.set(lastMessageL1BlockNumber);
+      void this.#lastL1BlockMessages.set(lastMessageL1BlockNumber);
 
       for (const message of messages) {
         if (message.index >= this.#l1ToL2MessagesSubtreeSize) {
           throw new Error(`Message index ${message.index} out of subtree range`);
         }
         const key = `${message.blockNumber}-${message.index}`;
-        void this.#newMessages.setIfNotExists(key, message.leaf.toBuffer());
-      }
-
-      return true;
-    });
-  }
-
-  /**
-   * Append new pending L1 to L2 messages to the store.
-   * @param messages - The L1 to L2 messages to be added to the store.
-   * @param l1BlockNumber - The L1 block number for which to add the messages.
-   * @returns True if the operation is successful.
-   */
-  addPendingMessages(messages: L1ToL2Message[], l1BlockNumber: bigint): Promise<boolean> {
-    return this.db.transaction(() => {
-      const lastL1BlockNumber = this.#lastL1BlockAddingMessages.get() ?? 0n;
-      if (lastL1BlockNumber >= l1BlockNumber) {
-        return false;
-      }
-
-      void this.#lastL1BlockAddingMessages.set(l1BlockNumber);
-
-      for (const message of messages) {
-        const entryKey = message.entryKey?.toString();
-        if (!entryKey) {
-          throw new Error('Message does not have an entry key');
-        }
-
-        void this.#messages.setIfNotExists(entryKey, {
-          message: message.toBuffer(),
-          fee: message.fee,
-          confirmed: false,
-        });
-
-        void this.#pendingMessagesByFee.update([message.fee, entryKey], 1);
-      }
-
-      return true;
-    });
-  }
-
-  /**
-   * Remove pending L1 to L2 messages from the store (if they were cancelled).
-   * @param entryKeys - The entry keys to be removed from the store.
-   * @param l1BlockNumber - The L1 block number for which to remove the messages.
-   * @returns True if the operation is successful.
-   */
-  cancelPendingMessages(entryKeys: Fr[], l1BlockNumber: bigint): Promise<boolean> {
-    return this.db.transaction(() => {
-      const lastL1BlockNumber = this.#lastL1BlockCancellingMessages.get() ?? 0n;
-      if (lastL1BlockNumber >= l1BlockNumber) {
-        return false;
-      }
-
-      void this.#lastL1BlockCancellingMessages.set(l1BlockNumber);
+        void this.#l1ToL2Messages.setIfNotExists(key, message.leaf.toBuffer());
 
-      for (const entryKey of entryKeys) {
-        const messageCtx = this.#messages.get(entryKey.toString());
-        if (!messageCtx) {
-          throw new Error(`Message ${entryKey.toString()} not found`);
-        }
-
-        void this.#pendingMessagesByFee.update([messageCtx.fee, entryKey.toString()], -1);
+        const indexInTheWholeTree =
+          (message.blockNumber - BigInt(INITIAL_L2_BLOCK_NUM)) * BigInt(NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP) +
```

Comment:
Cheeky breeky. As long as the size is not going to change across upgrades should be fine 👍